### DISCUSSION TITLE
How can I keep the diff of poetry.lock small?

### DISCUSSION THREAD

**Discussion by martin-thoma** [python-poetry/poetry#6949] [Q&A] [answered]
I regularly update the packages like this:

1. `poetry update`
2. Create a PR
3. Review the diff
4. Look up the changes if there was a major version bump
5. Wait for CI

This question is about step (3).

Typically the diff is small and easy to read. But once in a while I get like 1000 lines changed. And not because so many packages updated, but because of irrelevant changes. I also have the impression that those changes jumps back-and-forth (might be due to dependabot?)

Is it possible to get rid of those, either by using a poetry flag or by a package that modifies the `poetry.lock` file?

# Irrelevant changes in poetry.lock

Here are examples what I've seen.

## 1. Order of extra depencencies changed

![image](https://user-images.githubusercontent.com/80222180/199530857-b0ae374f-0f18-4295-8000-a886d2f96e26.png)

## 2. Lower/upper case changed

![image](https://user-images.githubusercontent.com/80222180/199531017-d63847ba-f089-4e8b-8746-4c2644e1c361.png)

> All comparisons of distribution names MUST be case insensitive, and MUST consider hyphens and underscores to be equivalent.

https://stackoverflow.com/a/31585483/562769

## 3. Hyphen / underscore change

![image](https://user-images.githubusercontent.com/80222180/199531318-d13f6dd7-1ccc-4dce-a6db-fb6c93c578e0.png)

**Comment by sbrunner** [comment]
It looks like that you use a newer version of Poetry or Poetry-core that with add some normalization from the PEP-517

**Comment by radoering** [comment, marked as answer]
The irrelevant changes are caused by using different versions of poetry:

- Ordered extras were introduced in poetry 1.1.15 in order to keep the diff small. 😉
- The lower/upper case change was introduced in poetry 1.2 to keep the lock file mostly compatible between 1.1 and 1.2.
- The hyphen/underscore change was introduced in poetry 1.2.2 to fix a bug with unnormalized extras.

To keep the diff small I'd recommend to always use the same version of poetry for locking. When switching to a newer version of poetry a larger diff cannot always be avoided.

**Reply by martin-thoma** [reply]
Is there a way to update the poetry.lock file to the new format without doing any package updates?

**Reply by finswimmer** [reply]
`poetry lock --no-update`

If this results in any update, than that's due to bug fixes in the locker.

**Comment by nils-borrmann-y42** [comment]
Poetry diffs are very verbose and often hard to make sense of. To keep track of what my dependencies are I have written a Github Action that will post a nicely readable summary of all changes to all pull requests. Maybe you will find this helpful: https://github.com/marketplace/actions/diff-poetry-lock

![image](https://user-images.githubusercontent.com/98901461/224825761-2a3f93ae-9259-4c6a-b37a-2ec061203a20.png)
